### PR TITLE
De hashed integration

### PR DESCRIPTION
**Pull request by gal-berger**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/23498

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by gal-berger** [comment]
@graphicool can you please help with the logo? https://www.dehashed.com

**Comment by graphicool** [comment]
@gal-berger i asked ayal in your behalf, please communicate on slack (he does not have an access to this github)

File: Packs/DeHashed/Integrations/DeHashed/DeHashed.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    """
+    if results_from:
+        results_from_int = convert_string_to_int(results_from)
+    else:
+        results_from_int = RESULTS_FROM
+    if results_to:
+        results_to_int = convert_string_to_int(results_to)
+    else:
+        results_to_int = RESULTS_TO
+
```
Comment (yuvalbenshalom) [inline Packs/DeHashed/Integrations/DeHashed/DeHashed.py:134]: lets have a check here that `results_from_int` and `results_to_int` are valid.
that means that from < to, that they are not negative, and they are in the page.
and return the correct errors.

File: Packs/DeHashed/Integrations/DeHashed/DeHashed.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    """
+    this command returns data regarding a compromised assets given as arguments
+    :param client: Demisto client
+    :param args:
+    - asset_type: email, ip_address, username, hashed_password, name, vin, address, phone,all_fields.
+    - value: value to search
+    - operation: choose a search type that you to perform.
+    - results_page_number: number for the next result page.
+    - results_from: sets results' start range
+    - results_to: sets results' end range
```
Comment (yuvalbenshalom) [inline Packs/DeHashed/Integrations/DeHashed/DeHashed.py:148]:
```suggestion
    - results_from: sets result's start range
    - results_to: sets result's end range
```

File: Packs/DeHashed/Integrations/DeHashed/DeHashed.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      required: true
+      secret: false
+    - default: false
+      description: number for the next result page. every page contains 5,000 records.
+      isArray: false
+      name: page
+      required: false
+      secret: false
+    - default: false
+      description: sets results' start range
```
Comment (yuvalbenshalom) [inline Packs/DeHashed/Integrations/DeHashed/DeHashed.yml:76]: we need to give more explanation here. lets talk about this.

File: Packs/DeHashed/Integrations/DeHashed/DeHashed.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      isArray: false
+      name: results_from
+      required: false
+      secret: false
+    - default: false
+      description: sets results' end range
```
Comment (yuvalbenshalom) [inline Packs/DeHashed/Integrations/DeHashed/DeHashed.yml:82]: we need to give more explanation here. lets talk about this.

File: Packs/DeHashed/Integrations/DeHashed/DeHashed.yml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+      type: String
+    - contextPath: DeHashed.Search.IpDddress
+      description: Object's ip address
+      type: Number
+    - contextPath: DeHashed.Search.Phone
+      description: Object's phone
+      type: Number
+    - contextPath: DeHashed.Search.ObtainedFrom
+      description: Object's source
+      type: String
```
Comment (yuvalbenshalom) [inline Packs/DeHashed/Integrations/DeHashed/DeHashed.yml:124]: the query outputs that we agreed upon should also be added here

File: Packs/DeHashed/Integrations/DeHashed/DeHashed.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+            query_value = " OR ".join(value)
+            if len(value) > 1:
+                query_value = f"({query_value})"
+
+        elif operation == "regex":
+            query_value = " ".join((f"/{value}/" for value in value))
+        if asset_type == "all_fields":
+            query_string = f"{query_value}"
+        else:
+            query_string = f"{asset_type}:{query_value}"
```
[the remark this answers is not in the document]
Comment (gal-berger) [reply Packs/DeHashed/Integrations/DeHashed/DeHashed.py:56]: search a value without specifying a asset type (like email, name...)

**Comment by kirbles19** [comment]
@gal-berger @yuvalbenshalom Done. Please regenerate the README from the updated YML.

File: Packs/DeHashed/Integrations/DeHashed/DeHashed.yml [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    - contextPath: Dehashed.LastQuery.DisplayedResults
+      description: The number of entries that were displayed in Cortex XSOAR from the last
+        query.
+      type: Number
+  dockerimage: demisto/python3:3.8.3.9324
+  feed: false
+  isfetch: false
+  longRunning: false
+  longRunningPort: false
+  runonce: false
```
Comment (yuvalbenshalom) [inline Packs/DeHashed/Integrations/DeHashed/DeHashed.yml:141]: please add fromversion: 5.5.0

**Pull request merged**